### PR TITLE
chore: Add Github Actions Workflow to build if Dockerfile updated

### LINKED ISSUE #291: Add GH Actions which will build Dockerfile if updated
GH Actions for pull-requests which runs `docker build` when `Dockerfile` changes and run the smoke test?

No need to push it to GHCR but just getting versions should be enough. Similar to this one - https://github.com/antonbabenko/pre-commit-terraform/blob/master/.github/workflows/build-image.yaml

_Originally posted by @antonbabenko in https://github.com/antonbabenko/pre-commit-terraform/issues/288#issuecomment-986849696_

### PR DESCRIPTION
**Pull request by mohitsaxenaknoldus**
Put an `x` into the box if that apply:

- [ ] This PR introduces breaking change.
- [ ] This PR fixes a bug.
- [x] This PR adds new functionality.
- [ ] This PR enhances existing functionality.

### Description of your changes

Closes #291 
Still need to add smoke tests.

### How has this code been tested

On my local branch.

### CODE REVIEW

**Review by MaxymVlasov — APPROVED** [review]
LGTM

File: .github/workflows/build-image-test.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - name: Check if Dockerfile changed
+        run: |
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            if [ $file = "Dockerfile" ]; then
+              echo "DOCKERFILE_CHANGED=true" >> $GITHUB_ENV
+            fi
+          done
+
+      - name: Build
+        if: env.DOCKERFILE_CHANGED == 'true'
```
Comment (antonbabenko) [inline .github/workflows/build-image-test.yaml:27]: This `if` can be simplified by changing `changed-files` step to support `with: files:` with `Dockerfile` (see https://github.com/tj-actions/changed-files#example step `changed-files-specific`).

There is no need to have `Check if Dockerfile changed` step, too.

**Comment by MaxymVlasov** [comment]
Well, good tests can be found in #319. 
I mean `container-structure-test-config` and `dive-ci`. I think they should be added here.

**Comment by MaxymVlasov** [comment]
@balihb can you please add `container-structure-test-config` and `dive-ci` from your recent PR to `build-image-test.yaml` by PR to `mohitsaxenaknoldus:issue291`?

**Comment by MaxymVlasov** [comment]
Or it will be too complicated and more easily to merge this PR and then make next to `master` 🤔 
@antonbabenko ?

File: .github/workflows/build-image-test.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          build-args: |
+            INSTALL_ALL=true
+          platforms: linux/amd64
+          push: false
+          tags: |
+            ghcr.io/${{ github.repository }}:${{ env.IMAGE_TAG }}
+
```
Comment (MaxymVlasov) [inline .github/workflows/build-image-test.yaml:32]:
```suggestion
```

**Comment by balihb** [comment]
> @balihb can you please add `container-structure-test-config` and `dive-ci` from your recent PR to `build-image-test.yaml` by PR to `mohitsaxenaknoldus:issue291`?

I can do it as a new PR.

**Review by antonbabenko — APPROVED** [review]
LGTM. Maybe update the PR title since it does not actually "run smoke tests"?

**Pull request merged**